tests/codegen/data_instrumentation_test.py

```diff
@@ -212,6 +213,34 @@ def dinstr(A: dace.float64[2 * N, 20]):
     assert np.allclose(dreport['tmp'], A + 1)
 
 
+def test_dinstr_in_loop_conditional():
+    @dace.program
+    def dinstr(A: dace.float64[20]):
+        tmp = np.copy(A)
+        for i in range(20):
+            tmp[i] = np.sum(tmp)
+        return tmp
+
+    sdfg = dinstr.to_sdfg(simplify=True)
+
+    # Set instrumentation on all access nodes
+    for node, _ in sdfg.all_nodes_recursive():
+        if isinstance(node, nodes.AccessNode):
+            node.instrument = dace.DataInstrumentationType.Save
+            node.instrument_condition = CodeBlock('i == 0', language=dace.Language.CPP)
```

Comment:
For completeness, please add a test with python codeblocks and “i ** 2 == 4”

Reply:
Done